flow/connectors/eventhub/eventhub.go

```diff
@@ -80,11 +85,104 @@ func (c *EventHubConnector) SetupMetadataTables(_ context.Context) error {
 	return nil
 }
 
+func lvalueToEventData(ls *lua.LState, value lua.LValue) (ScopedEventhubData, error) {
+	var scoped ScopedEventhubData
+	switch v := value.(type) {
+	case lua.LString:
+		scoped.Data = &azeventhubs.EventData{
+			Body: shared.UnsafeFastStringToReadOnlyBytes(string(v)),
+		}
+	case *lua.LTable:
+		value, err := utils.LVAsReadOnlyBytes(ls, ls.GetField(v, "value"))
+		if err != nil {
+			return scoped, fmt.Errorf("invalid value, %w", err)
+		}
+		destination, err := utils.LVAsStringOrNil(ls, ls.GetField(v, "destination"))
+		if err != nil {
+			return scoped, fmt.Errorf("invalid destination, %w", err)
+		}
+		if destination != "" {
+			scoped.Hub, err = NewScopedEventhub(destination)
+			if err != nil {
+				return scoped, fmt.Errorf("invalid topic, %w", err)
+			}
+		}
+		namespace, err := utils.LVAsStringOrNil(ls, ls.GetField(v, "namespace"))
+		if err != nil {
+			return scoped, fmt.Errorf("invalid namespace, %w", err)
+		}
+		if namespace != "" {
+			scoped.Hub.NamespaceName = namespace
+		}
+		partCol, err := utils.LVAsStringOrNil(ls, ls.GetField(v, "partitionColumn"))
+		if err != nil {
+			return scoped, fmt.Errorf("invalid partitionColumn, %w", err)
+		}
+		if partCol != "" {
+			scoped.Hub.PartitionKeyColumn = partCol
+		}
+		key, err := utils.LVAsStringOrNil(ls, ls.GetField(v, "key"))
```

Comment:
```suggestion
		key, err := utils.LVAsStringOrNil(ls, ls.GetField(v, "partitionKey"))
```
To make it more clear

Reply:
Using "key" to be consistent with kafka/pubsub